ocaml/src/alba_eviction.ml

```diff
@@ -108,66 +108,106 @@ let do_eviction
       delete_empty_namespaces ();
     ]
 
-let alba_eviction
-      (alba_client : Alba_base_client.client)
-      ~get_prefix_preset_pairs
-  =
-  let coordinator =
-    Maintenance_coordination.make_maintenance_coordinator
-      (alba_client # mgr_access)
-      ~lease_name:"cache_eviction"
-      ~lease_timeout:20.
-      ~registration_prefix:"cache_eviction"
-  in
-  let () = coordinator # init in
+let should_evict (alba_client : Alba_base_client.client) coordinator =
+  if coordinator # is_master
+  then
+    begin
+      let cnt_total, cnt_full, _ =
+        Hashtbl.fold
+          (fun osd_id (osd_info, osd_state, _)
+               (cnt_total, cnt_full, acc) ->
+           let open Nsm_model in
+           let used = Int64.to_float osd_info.OsdInfo.used in
+           let total = Int64.to_float osd_info.OsdInfo.total in
+           let fill_ratio = used /. total in
+           (cnt_total + 1,
+            cnt_full + (if fill_ratio > 0.9 then 1 else 0),
+            (osd_id,
+             used, total,
+             fill_ratio)
+            :: acc))
+          (alba_client # osd_access # osds_info_cache)
+          (0, 0, [])
+      in
+      (* more than half of the disks are >90% filled,
+       * so let's do some cleanup *)
+      float cnt_total /. float cnt_full < 2.
+    end
+  else
+    false
 
-  Lwt_extra2.run_forever
-    "alba eviction"
-    (fun () ->
+let lru_collect_some_garbage alba_client redis_client key =
+  let module R = Redis_lwt.Client in
+  R.zrangebyscore
+    redis_client
+    key
+    0. max_float
+    ~limit:(0, 100)
+  >>= fun items ->
+
+  let items =
+    List.map
+      (fun r ->
+       match r with
+       | `Bulk (Some b) -> b
```

Comment:
The items are added to redis in https://github.com/openvstorage/alba/pull/275/files#diff-044afbbce530f0adfcecda0062438fbdR100 .
`Bulk is just the way the ocaml redis library returns the results from zrangebyscore.